tests/select-input-test.js

```diff
@@ -0,0 +1,335 @@
+
+const updateValue = (formId, fieldName, fieldValue) => {
+  value = fieldValue
+}
+
+describe('<SelectInput />', function() {
+  const exampleLabel = 'Example Select Input'
+  const typeaheadInput = (
+    <SelectInput
+      formData={ exampleFormData }
+      formId='example-form'
+      label={ exampleLabel }
+      name='select-typeahead'
+      onChange={ updateValue }
+      options={ options }
+      value={ value }
+    />
+  )
+  const buttonInput = (
+    <SelectInput
+      formData={ exampleFormData }
+      formId='example-form'
+      label={ exampleLabel }
+      name='select-button'
+      onChange={ updateValue }
+      options={ options.map(opt => opt.value) }
+      typeAhead={ false }
+      value={ value }
+    />
+  )
+
+  // make sure there are separate divs to attach each to
+  const root1 = document.createElement('div')
+  root1.id = 'root-1'
+  const root2 = document.createElement('div')
+  root2.id = 'root-2'
+  document.getElementById('root').appendChild(root1)
+  document.getElementById('root').appendChild(root2)
+
+  const wrapper = Enzyme.mount(typeaheadInput, { attachTo: document.getElementById('root-1') })
+  const wrapper2 = Enzyme.mount(buttonInput, { attachTo: document.getElementById('root-2') })
+
+  function resetWrappers() {
+    wrapper.unmount()
+    wrapper.mount(typeaheadInput, { attachTo: document.getElementById('root-1') })
+
+    wrapper2.unmount()
+    wrapper2.mount(buttonInput, { attachTo: document.getElementById('root-2') })
+  }
+
+  describe('tests set up with root for mounting', function() {
+    it ('is ready to test', function() {
+      assert.ok(document.getElementById('root'))
+      assert.ok(document.getElementById('select-typeahead'))
+      assert.ok(document.getElementById('select-button'))
+    })
+  })
+
+  describe('rendering correctly:', function() {
+    it ('renders without breaking', function() {
+      assert.ok(sinon.spy(wrapper.instance(), 'componentDidMount'))
+      assert.ok(sinon.spy(wrapper2.instance(), 'componentDidMount'))
+    })
+
+    it ('renders the selected option\'s label if the option is an object', function() {
+      const typeaheadValue = wrapper.find('#select-typeahead-placeholder').instance().value
+      assert.equal(typeaheadValue, 'C')
+    })
+
+    it ('renders the selected option\'s value if the option is a string or number', function() {
+      const buttonValue = wrapper2.find('#select-button-placeholder').text()
+      assert.equal(buttonValue, 'c')
+    })
+
+    it ('renders the typeahead by default', function() {
+      const typeahead = wrapper.find('#select-typeahead-placeholder')
+      assert.equal(typeahead.type(), 'input')
+    })
+
+    it ('renders the select with no typeahead if `typeAhead` prop is false', function() {
+      const button = wrapper2.find('#select-button-placeholder')
+      assert.equal(button.type(), 'button')
+    })
+  })
+
+  describe('events trigger opening, closing, focusing etc. appropriately:', function() {
```

Comment:
Any code in here that repeats can be added to a beforeEach block nested in the describe.  IE - `resetWrappers()`